src/schematools/types.py

```diff
@@ -933,8 +944,106 @@ def validate(self, schema: DatasetSchema) -> None:
         table.validate(self.data)
 
 
+@total_ordering
+class PermissionLevel(Enum):
+    """The various levels that can be provided on specific fields."""
+
+    # Higher values give higher preference. The numbers are arbitrary and for internal usage
+    # allowing to test test "read > encoded" for example.
+    read = 50
+    encoded = 40
+    random = 30
+    letters = 10
+    subobjects_only = 1  # allows to open a table only to access sub-fields
+    none = 0  # means no permission.
+
+    highest = read
+
+    @classmethod
+    def from_string(cls, value: Optional[str]) -> PermissionLevel:
+        """Cast the string value to a permission level object."""
+        if value is None:
+            return cls.none
+        elif "_" in value:
+            # Anything with an underscore is internal
+            raise ValueError("Invalid permission")
+        else:
+            return cls[value]
+
+    def __str__(self):
+        # Using the name as official value.
+        return self.name
+
+    def __bool__(self):
+        """The 'none' level is recognized as "NO PERMISSION"."""
+        return bool(self.value)
```

Comment:
Do we use this anywhere?

Reply:
yes! It's used in every `if permission` check. So you can so a simple `if permission` and `if permission.level ...` checks.